ksqldb-engine/src/main/java/io/confluent/ksql/engine/generic/GenericExpressionResolver.java

```diff
@@ -0,0 +1,107 @@
+
+import com.google.common.collect.Iterables;
+import io.confluent.ksql.GenericRow;
+import io.confluent.ksql.execution.codegen.CodeGenRunner;
+import io.confluent.ksql.execution.codegen.ExpressionMetadata;
+import io.confluent.ksql.execution.expression.tree.Expression;
+import io.confluent.ksql.execution.expression.tree.NullLiteral;
+import io.confluent.ksql.execution.expression.tree.VisitParentExpressionVisitor;
+import io.confluent.ksql.function.FunctionRegistry;
+import io.confluent.ksql.logging.processing.ProcessingLogger;
+import io.confluent.ksql.logging.processing.RecordProcessingError;
+import io.confluent.ksql.name.ColumnName;
+import io.confluent.ksql.schema.ksql.DefaultSqlValueCoercer;
+import io.confluent.ksql.schema.ksql.LogicalSchema;
+import io.confluent.ksql.schema.ksql.SchemaConverters;
+import io.confluent.ksql.schema.ksql.SqlValueCoercer;
+import io.confluent.ksql.schema.ksql.types.SqlBaseType;
+import io.confluent.ksql.schema.ksql.types.SqlType;
+import io.confluent.ksql.util.KsqlConfig;
+import io.confluent.ksql.util.KsqlException;
+import java.util.Objects;
+import java.util.function.Supplier;
+import java.util.stream.Stream;
+
+/**
+ * Builds a Java object, coerced to the desired type, from an arbitrary SQL
+ * expression that does not reference any source data.
+ */
+class GenericExpressionResolver extends VisitParentExpressionVisitor<Object, Void> {
+
+  private static final Supplier<String> IGNORED_MSG = () -> "";
+  private static final ProcessingLogger THROWING_LOGGER = errorMessage -> {
+    throw new KsqlException(((RecordProcessingError) errorMessage).getMessage());
+  };
+
+  private final SqlType fieldType;
+  private final ColumnName fieldName;
+  private final LogicalSchema schema;
+  private final SqlValueCoercer sqlValueCoercer = DefaultSqlValueCoercer.INSTANCE;
+  private final FunctionRegistry functionRegistry;
+  private final KsqlConfig config;
+
+  GenericExpressionResolver(
+      final SqlType fieldType,
+      final ColumnName fieldName,
+      final LogicalSchema schema,
+      final FunctionRegistry functionRegistry,
+      final KsqlConfig config
+  ) {
+    this.fieldType = Objects.requireNonNull(fieldType, "fieldType");
+    this.fieldName = Objects.requireNonNull(fieldName, "fieldName");
+    this.schema = Objects.requireNonNull(schema, "schema");
+    this.functionRegistry = Objects.requireNonNull(functionRegistry, "functionRegistry");
+    this.config = Objects.requireNonNull(config, "config");
+  }
+
+  @Override
+  protected Object visitExpression(final Expression expression, final Void context) {
+    final ExpressionMetadata metadata =
+        Iterables.getOnlyElement(
+            CodeGenRunner.compileExpressions(
+                Stream.of(expression),
+                "insert value",
+                schema,
+                config,
+                functionRegistry)
+        );
+
+    // we expect no column references, so we can pass in an empty generic row
```

Comment:
It never gets to this point because it can't prepare the statement. Added a test to confirm the behavior, making it better though is a little out of scope for this PR:
```
    {
      "name": "should fail on column reference",
      "statements": [
        "CREATE STREAM TEST (K STRING KEY, ID INT) WITH (kafka_topic='test_topic', value_format='JSON');",
        "INSERT INTO TEST (K, ID) VALUES (FOO.bar, '10');"
      ],
      "expectedError": {
        "type": "io.confluent.ksql.rest.entity.KsqlStatementErrorMessage",
        "message": "Failed to prepare statement: 'FOO' is not a valid stream/table name or alias.",
        "status": 400
      }
    }
```